src/loops.rs

```diff
@@ -69,6 +72,37 @@ impl LintPass for LoopsPass {
                 }
             }
 
+            // if this for-loop is iterating over a two-sided range...
+            if let ExprRange(Some(ref start_expr), Some(ref stop_expr)) = arg.node {
```

Comment:
So it would seem that this doesn't catch (10..0).rev() - I added a unit test for this and related function calls on ranges and it would seem that they do not get linted as I expected when I wrote this code. I've got tests now but I'll take another look tomorrow.